ssl/record/rec_layer_s3.c

```diff
@@ -356,7 +356,7 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, size_t len,
      * promptly send beyond the end of the users buffer ... so we trap and
      * report the error in a way the user will notice
      */
-    if (len < s->rlayer.wnum) {
+    if ((len < s->rlayer.wnum) || ((tot != 0) && (len < (tot + s->rlayer.wpend_tot)))) {
```

Comment:
In that case we should check `wb->left != 0` because that is the test we use later on in this function immediately prior to the ssl3_write_pending() call.